include/aspect/volume_of_fluid/utilities.h

```diff
@@ -0,0 +1,152 @@
+     */
+    template<int dim>
+    void xFEM_Heaviside_derivative_d(const int degree,
+                                     const Tensor<1, dim, double> normal,
+                                     const double d,
+                                     const std::vector<Point<dim>> &points,
+                                     std::vector<double> &values);
+
+
+    /**
+     * Function to do Newton iteration calculation of correct d for a given
+     * normal to get vol_frac from xFEM_Heaviside integrated against the given
+     * weights.
+     *
+     * @param degree Maximum degree for exact integration
+     * @param normal Interface normal vector, pointed away from the included region
+     * @param vol_frac Cell volume fraction in physical space - used for target value for integration
+     * @param vol Cell volume in physical space - used for target value for integration
+     * @param epsilon Tolerance for Newton iteration
+     * @param points Quadrature points to use for update
+     * @param weights JxW values to use for quadrature
+     */
+    template<int dim>
+    double compute_interface_location_newton(const int degree,
+                                             const Tensor<1, dim, double> normal,
+                                             const double vol_frac,
+                                             const double vol,
+                                             const double epsilon,
+                                             const std::vector<Point<dim>> &points,
+                                             const std::vector<double> &weights);
+
+    /**
+     * Function to calculate volume fraction contained by indicator function
+     * $H(d-normal*(x'-x_{cen}'))$ on the $[0, 1]^dim$ unit cell where
+     * $x_{cen}$ is the unit cell center, using a polynomial mapping of degree
+     * up to "degree".
+     *
+     * @param degree Maximum degree for exact integration
+     * @param normal Interface normal vector, pointed away from the included region
+     * @param d Interface parameter specifying location of interface in unit cell by distance from cell center
+     * @param points Quadrature points to use for update
+     * @param weights JxW values to use for quadrature
+     */
+    template<int dim>
+    double compute_fluid_volume_xFEM(const int degree,
+                                     const Tensor<1, dim, double> normal,
+                                     const double d,
+                                     const std::vector<Point<dim>> &points,
+                                     const std::vector<double> &weights);
+
+    /**
+     * Function to calculate flux volume fraction based on a method of
+     * characteristics approximation of the interface on the cell's face over
+     * the timestep.
+     */
+    template<int dim>
+    double calc_volume_of_fluid_flux_edge (const unsigned int compute_direction,
```

Comment:
what is the meaning of `edge`? would `calculate_volume_flux` already capture all necessary information?